examples/ldap/main.tf

```diff
@@ -0,0 +1,105 @@
+resource "ibm_compute_vm_instance" "ldap" {
```

Comment:
This file is the same as the module file.  It needs to call the module and pass in the values for the variables.  You can hard code most things, but allow user to set hostname, domain, datacenter.  

```
provider "ibm" {
  version          = "~> 1.12"
  ibmcloud_api_key = var.ibmcloud_api_key
}

data "ibm_resource_group" "group" {
  name = var.resource_group
}

resource "null_resource" "mkdir_kubeconfig_dir" {
  triggers = { always_run = timestamp() }
  provisioner "local-exec" {
    command = "mkdir -p ${local.config_dir}"
  }
}

data "ibm_container_cluster_config" "cluster_config" {
  depends_on = [null_resource.mkdir_kubeconfig_dir]
  cluster_name_id   = var.cluster_id
  resource_group_id = data.ibm_resource_group.group.id
  config_dir        = local.config_dir
}

// Module:

module "ldap" {
  source = "../../modules/ldap"
  enable = true
  region              = var.region
hostname             = var.hostname
  domain               = var.ibmcloud_domain
  ssh_key_ids        = var.ssh_key_ids
  os_reference_code    = var.os_reference_code
  datacenter           = var.datacenter
  network_speed    = 10
  hourly_billing       = true
  private_network_only = false
  cores                = var.cores
  memory           = var.memory
  disks                = var.disks
  local_disk        = false
}
```